Makefile

```diff
@@ -14,10 +14,25 @@ CMAKE_EXTRA_FLAGS ?=
 # CMAKE_INSTALL_PREFIX
 #   - May be passed directly or as part of CMAKE_EXTRA_FLAGS.
 #   - `checkprefix` target checks that it matches the CMake-cached value. #9615
+ifeq (,$(CMAKE_EXTRA_FLAGS))
+CMAKE_INSTALL_PREFIX ?=
+else
 CMAKE_INSTALL_PREFIX ?= $(shell echo $(CMAKE_EXTRA_FLAGS) | 2>/dev/null \
     grep -o 'CMAKE_INSTALL_PREFIX=[^ ]\+' | cut -d '=' -f2)
+endif
 ifneq (,$(CMAKE_INSTALL_PREFIX))
-  CMAKE_FLAGS += -DCMAKE_INSTALL_PREFIX=$(CMAKE_INSTALL_PREFIX)
+  CMAKE_EXTRA_FLAGS += -DCMAKE_INSTALL_PREFIX=$(CMAKE_INSTALL_PREFIX)
+
+checkprefix:
+	@if [ -f build/.ran-cmake ]; then \
+	  cached_prefix=$(shell $(CMAKE_PRG) -L -N build | 2>/dev/null grep 'CMAKE_INSTALL_PREFIX' | cut -d '=' -f2); \
+	  if ! [ "$(CMAKE_INSTALL_PREFIX)" = "$$cached_prefix" ]; then \
+	    echo "Re-running CMake for changed CMAKE_INSTALL_PREFIX."; \
```

Comment:
the old message should be kept as it was:

    printf "\nCMAKE_INSTALL_PREFIX '$(CMAKE_INSTALL_PREFIX)' does not match cached value '%s'\n" "$$cached_prefix"; \
    printf "Re-running CMake\n"

Reply:
Used a single line, without leading newline.  Ok?

Reply:
` - re-running CMake.` at the end of the line is hard to see. I crafted the original form specifically for readability. `printf "Re-running CMake\n"` should be a separate line.

Reply:
How about prefixing the line then?

> "Re-running CMake: CMAKE_INSTALL_PREFIX …"

I found it confusing with what ninja would say when detecting changes ("Re-running CMake..").

Reply:
I think it is good to have this in a single line to indicate action and reason belong together.

Reply:
Then indent it, as the original message already did...

Reply:
current form looks good.